tools/aisbench.py

```diff
@@ -0,0 +1,220 @@
+    }
+    DATASET_RENAME = {
+        "aime2024": "aime",
+        "gsm8k-lite": "gsm8k",
+        "textvqa-lite": "textvqa"
+    }
+
+    def _run_aisbench_task(self):
+        dataset_conf = self.dataset_conf.split('/')[-1]
+        if self.task_type == "accuracy":
+            aisbench_cmd = [
+                'ais_bench', '--models', f'{self.request_conf}_custom',
+                '--datasets', f'{dataset_conf}', '--debug'
+            ]
+        if self.task_type == "performance":
+            aisbench_cmd = [
+                'ais_bench', '--models', f'{self.request_conf}_custom',
+                '--datasets', f'{dataset_conf}_custom', '--debug', '--mode',
+                'perf'
+            ]
+            if self.num_prompts:
+                aisbench_cmd.extend(['--num-prompts', str(self.num_prompts)])
+        print(f"running aisbench cmd: {' '.join(aisbench_cmd)}")
+        self.proc: subprocess.Popen = subprocess.Popen(aisbench_cmd,
+                                                       stdout=subprocess.PIPE,
+                                                       stderr=subprocess.PIPE,
+                                                       text=True)
+
+    def __init__(self,
+                 model: str,
+                 port: int,
+                 aisbench_config: dict,
+                 verify=True):
+        self.result_line = None
+        self.dataset_path = snapshot_download(aisbench_config["dataset_path"],
+                                              repo_type='dataset')
+        self.task_type = aisbench_config["case_type"]
+        self.request_conf = aisbench_config["request_conf"]
+        self.dataset_conf = aisbench_config.get("dataset_conf")
+        self.num_prompts = aisbench_config.get("num_prompts")
+        self.max_out_len = aisbench_config["max_out_len"]
+        self.batch_size = aisbench_config["batch_size"]
+        self.request_rate = aisbench_config.get("request_rate", 0)
+        self.model = model
+        self.model_path = snapshot_download(model)
+        self.port = port
+        self.exp_folder = None
+        self._init_dataset_conf()
+        self._init_request_conf()
+        self._run_aisbench_task()
+        self._wait_for_task()
+        if verify:
+            self.baseline = aisbench_config.get("baseline", 1)
+            if self.task_type == "accuracy":
+                self.threshold = aisbench_config.get("threshold", 1)
+                self._accuracy_verify()
+            if self.task_type == "performance":
+                self.threshold = aisbench_config.get("threshold", 0.97)
+                self._performance_verify()
+
+    def _init_dataset_conf(self):
+        if self.task_type == "accuracy":
+            dataset_name = os.path.basename(self.dataset_path)
+            dataset_rename = self.DATASET_RENAME.get(dataset_name, "")
+            dst_dir = os.path.join(DATASET_DIR, dataset_rename)
+            command = ["cp", "-r", self.dataset_path, dst_dir]
```

Comment:
This will cause redundant dataset copy, not sure why we can't use modelscope one directly?